### PR TITLE
Added icon details view

### LINKED ISSUE #17: Add compact layout
**:information_source: If you're interested in this feature, please leave comment or add a :+1:**

---

### Kind of issue 

- [x] Feature
- [ ] Bug
- [x] Design
- [ ] Other, namely:

### Description

Following [the original design](https://github.com/simple-icons/simple-icons/issues/980#issuecomment-427185048), a "compact layout" mode can be added to the website. The end-design may differ from what is displayed there, it is just a guideline.

### LINKED ISSUE #43: [Suggestion] Add "Icon detail" view
**:information_source: If you're interested in this feature, please leave comment or add a :+1:**

---

### Kind of issue 

- [x] Feature
- [ ] Bug
- [ ] Design
- [ ] Other, namely:

### Description

As per [this comment](https://github.com/simple-icons/simple-icons/issues/980#issuecomment-547591284), add an "icon detail view". This would display a particular icon at a larger size and potentially give download options such as #31, #32, #33 for that particular icon, as seen in [this example mockup](https://user-images.githubusercontent.com/10723208/67801718-12d6ab00-fa8a-11e9-8422-fb5790561de9.png).

I'm not sure how useful this is. The larger icon can be nice, but if that is the only benefit I'm not sure something like the mockup below is suitable. The icon-specific download options can be nice, but I would expect that everyone would nearly always want the same download option for all icons.

What it may allow us to do is things like:

- Show everything that is already in the card (the SVG, color, brand guidelines URL, license name & URL)
- Show aliases (https://github.com/simple-icons/simple-icons/issues/4797)
- Show the SVG source URL
- Show the brand website URL
- Show if the icon will be removed in next versions
- Link to milestone and pull request of icon removal, if will be removed in next versions
- Option to copy the brand name
- Option to copy the brand slug
- Option to copy the font class
- Option to copy the font code point
- Option to copy the entire SVG (even though that is already possible)
- Option to copy the path data
- Option to copy code to use in NPM
- Download svgs in the brand color (https://github.com/simple-icons/simple-icons-website/issues/32)
- Show the icon in a variety of sizes (https://github.com/simple-icons/simple-icons-website/issues/40)
- More? (Suggestions are welcome)

### LINKED ISSUE #95: Add link to report outdated icons
### Kind of issue 

- [X] Feature
- [ ] Bug
- [ ] Design
- [ ] Other, namely:

We should make it easier for users to report outdated icons in an effort to keep our collection up-to-date. Based on discussion (see below) we think the link should basically just point to [the Icon update template](https://github.com/simple-icons/simple-icons/issues/new?assignees=&labels=icon+outdated&template=icon_update.md) as (a) this reduces the chances (prevents?) of someone creating many issues quickly in an automated way, and (b) while prefilling details such as the brand name is _nice_, it doesn't add to much value to the UX while adding a lot of HTML (at least one element per icon = at least 1900 elements).

### LINKED ISSUE #149: [Suggestion] Option to download icon with the color information baked in
### Kind of issue

- [x] Feature
- [ ] Bug
- [ ] Design
- [ ] Other, namely:

### Description

The icons are useful beyond web applications. Without an HTML context, it's not possible to specify the fill color.

It would be great to make the SVGs downloadable with the color baked in.

### PR DESCRIPTION
**Pull request by jorgeamadosoria**
Addresses: 
#17 - Compact layout 
#95 - Link to outdated icons
#43 - Icon detail view
#149 - Download of color SVGs

image

The report button simply links to the issue template for update, and fills the title with the brand to be updated. It also includes a test in the e2e file.

-----------------------------------------
The changes for the compact layout are:
- Reduced gutters
- Reduced width for cards
- Reduced margins for card title
- Changed the color button to vertical to reduce the width of the card
- Removed the leading # in the color button text to reduce the width of it
- Set titles to a fixed width with an ellipsis overflow to avoid multilineal titles (full title available as tooltip)
- Removed guidelines and licenses from the cards.

this change should be complemented with an additional "details" button alongside the other two. In this detail view the guidelines and licenses will be shown, to avoid loss of information

### CODE REVIEW

File: public/index.pug
Comment (mondeja) [inline public/index.pug]: Not would be more straightforward to just set `Update <> icon` in the title? Not sure, I'm not a native english speaker.

```suggestion
                href=`https://github.com/simple-icons/simple-icons/issues/new?labels=icon+outdated&template=icon_update.md&title=Update%20${icon.title}%20icon`,
```

Also, [Potential Duplicates workflow](https://github.com/simple-icons/simple-icons/blob/b7375ec336e71023f19193edaacfb108ee2bd551/.github/workflows/potential-duplicates.yml) must be updated adding `Update` and / or `for` words.

File: public/index.pug
Comment (mondeja) [inline public/index.pug]: Could you address it in this PR? It would be strange to remove useful functionality in the next release.

File: public/index.pug
Comment (mondeja) [inline public/index.pug]: Is missing a title here?

```suggestion
                role='button'
                title=`Report ${icon.title} as outdated`
```

**Review by mondeja — CHANGES_REQUESTED** [review]
Great.

As I understand, [in the original report](https://github.com/simple-icons/simple-icons/issues/980#issuecomment-427185048), the reporter mocked a toggling functionality to change between layouts.

Maybe that would be even enough to solve the problem of the lack of licenses and brands guidelines in this: a toggling feature just like order, color and download type buttons. The licenses and brand guidelines would be only shown on expanded layout. Not sure about this, but at least the licenses and brand guidelines functionality would not be dropped in this PR, which I consider a setback.

Also, I think that the design currently implemented in this PR could be a bit claustrophobic for some users.

**Comment by jorgeamadosoria** [comment]
> Great.
> 
> As I understand, [in the original report](https://github.com/simple-icons/simple-icons/issues/980#issuecomment-427185048), the reporter mocked a toggling functionality to change between layouts.
> 
> Maybe that would be even enough to solve the problem of the lack of licenses and brands guidelines in this: a toggling feature just like order, color and download type buttons. The licenses and brand guidelines would be only shown on expanded layout. Not sure about this, but at least the licenses and brand guidelines functionality would not be dropped in this PR, which I consider a setback.
> 
> Also, I think that the design currently implemented in this PR could be a bit claustrophobic for some users.

a toggle was floated, but I seem to remember that the compact layout was ultimately just decided to be the layout. @ericcornelissen had some comments to that effect, I remember. I may be wrong, I would have to go back and check, but I'm pretty sure the toggle was dropped.

More importantly, I (personally) don't think a toggle is all that important. To the point that if the previous layout was to be kept as an option, I wouldn't have taken this issue. It would be complicating things for not a lot of improvement.

As for license and guidelines, the problem is that this PR is not really for merging just yet. I want to validate the design first, hence why it's here. Once that's done, I will build the detail view PR upon this one.

The detail view PR that is under way should build upon the compact layout, and will include license and guidelines. No need for a separate toggle for it. We could have either a toggle for standard layout / compact layout, or no toggle at all and just compact layout (my choice is the latter). 

I agree that the current design may be claustrophobic, specially compared with the previous one. I personally like it, but I'm biased so my opinion does not carry a lot of weight. I'm willing to redesign this if there are other, more aesthetically pleasing ideas. We need to just keep in mind that the design has to be compact, but other than that, I'm open to ideas for improvement.

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: I didn't miss it, I just didn't add it.
If we were to add it, I would prefer 'Report ${icon.title} as outdated'

Can you change the suggestion so we can commit it?

File: public/index.pug
Comment (jorgeamadosoria) [reply public/index.pug]: No, this was removed on purpose. Look at my answer below. This PR is not to be committed just yet. The detail view PR will build upon this one, and that one will show license and guidelines. The information won't be lost after merging, since we will be merging the layout and detail view together.

File: public/index.pug
Comment (mondeja) [reply public/index.pug]: Done 🔧

**Comment by jorgeamadosoria** [comment]
Now going over the original thread, I can't find the place the toggle was discussed for dropping. Maybe it never was and I imagined the whole thing. 
In any case, is a choice of layouts really necessary?

**Comment by mondeja** [comment]
> In any case, is a choice of layouts really necessary?

I think that could be great, even necessary due to a lot of things, but this would also depend on how to details view is solved.

- The issue https://github.com/simple-icons/simple-icons-pdf/issues/7 could be a great addition for the website. Display new last updated date of an icon and the date of inclusion is really useful. This probably would be too much information for a details view in compact mode, but it would fit in a details view for expanded mode. Even could be added in expanded mode without "details view", I don't know because I still don't know what you mean by a detail view.
- Many users could feel rejected by a compact mode as the unique option, especially non-developer users who go for simple icons to only download an icon to play with it. The first impression of such a reduced view could scare them away. Yesterday tested this compact mode with an acquaintance and it found it too overwhelming in a laptop, not tested in a mobile view though.

> It would be complicating things for not a lot of improvement.

:thinking: It is just another theme with `media` selectors, like dark-light themes, changing 11 CSS properties. I don't think it's too much and it doesn't introduce any new patterns to the project. Is not an improvement, but it would not regress at least.

**Comment by jorgeamadosoria** [comment]
> > In any case, is a choice of layouts really necessary?
> 
> I think that could be great, even necessary due to a lot of things, but this would also depend on how to details view is solved.
> 
> * The issue [Set Creation Date of PDF to date of addition / modification in main repo. simple-icons-pdf#7](https://github.com/simple-icons/simple-icons-pdf/issues/7) could be a great addition for the website. Display new last updated date of an icon and the date of inclusion is really useful. This probably would be too much information for a details view in compact mode, but it would fit in a details view for expanded mode. Even could be added in expanded mode without "details view", I don't know because I still don't know what you mean by a detail view.
> * Many users could feel rejected by a compact mode as the unique option, especially non-developer users who go for simple icons to only download an icon to play with it. The first impression of such a reduced view could scare them away. Yesterday tested this compact mode with an acquaintance and it found it too overwhelming in a laptop, not tested in a mobile view though.
> 
> > It would be complicating things for not a lot of improvement.
> 
> 🤔 It is just another theme with `media` selectors, like dark-light themes, changing 11 CSS properties. I don't think it's too much and it doesn't introduce any new patterns to the project. Is not an improvement, but it would not regress at least.

Detail views ticket: #43 
It's a modal / page / slide that has more details about the icons than what is shown in the card. Everything that you have can be shown in a detail view.

I wouldn't make detail view compact / comfortable. Detail view is supposed to contain everything, not just a bit of info, so making it compact kind of defeats the purpose.

I'm against the toggle between layouts. It's more code for a user feature than I'm not sure if it's going to see a lot of use. If the current compact layout is overwhelming due to having too much going on, then we can close the PR and forget about the compact layout, or we could tweak the compact layout to make it less aggressive. After all, the website, even with compact layout mode, is already very generous with the size of icon names and previews, when compared with other icon libraries.

Also note that compact layout not only changes CSS properties, it also requires some markup changes. Maybe those could be dealt away with more CSS, I don't know. I can try to make the compact layout work with only CSS to remove that variable from the discussion.

It's not that it is not doable: it is. I just don't know if it's something we really need to have.

**Comment by mondeja** [comment]
OK, I will not block this as the request is not clear about it, so go ahead with compact layout only :+1: We can always add the toggle if any maintainer or user asks for it.

**Comment by jorgeamadosoria** [comment]
> OK, I will not block this as the request is not clear about it, so go ahead with compact layout only 👍 We can always add the toggle if any maintainer or user asks for it.

As a compromise, I'm going to try and make the changes CSS only, so the toggle can be added without much effort, as you pointed out before.

In any case, this PR is not going to be merged any time soon: I need to add the detail view first or we would be losing information. So we have time to get more feedback either way.

**Comment by jorgeamadosoria** [comment]
@mondeja Hey, so, I moved the changes in the HTML to be handled by CSS only, and it turned out to be easier than I thought. So I went ahead and added the toggle you wanted. 

I still think we don't need that, but for what it's worth, is in there now.

File: public/scripts/layout.js
Comment (mondeja) [inline public/scripts/layout.js]: The same than #157

```suggestion
  } else {
    storage.setItem(STORAGE_KEY_LAYOUT, DEFAULT_LAYOUT);
  }
```

File: scripts/removed-icons.js
Comment (mondeja) [inline scripts/removed-icons.js:1]: This file would be in another PR right?

File: public/scripts/layout.js
Comment (mondeja) [inline public/scripts/layout.js]: The third conditional has no sense here.

```suggestion
    } else if (selected === LAYOUT_COMPACT) {
      $body.classList.add(LAYOUT_COMPACT);
      $body.classList.remove(LAYOUT_COMFORTABLE);
    }
```

File: tests/e2e.test.js
Comment (mondeja) [inline tests/e2e.test.js:141]: Also this, should be in another PR.

File: webpack.config.js
Comment (mondeja) [inline webpack.config.js]: This changes also.

File: public/removed.pug
Comment (mondeja) [inline public/removed.pug]: This file too, in another PR.

**Review by mondeja — DISMISSED** [review]

File: tests/e2e.test.js
Comment (jorgeamadosoria) [reply tests/e2e.test.js:141]: no, this is supposed to be here. It's "compact layout and report outdated icon" together.

File: public/removed.pug
Comment (jorgeamadosoria) [reply public/removed.pug]: yes, this is here by mistake.

File: webpack.config.js
Comment (jorgeamadosoria) [reply webpack.config.js]: yes, this is here by mistake.

File: scripts/removed-icons.js
Comment (jorgeamadosoria) [reply scripts/removed-icons.js:1]: yes, this is here by mistake.

File: public/index.pug
Comment (mondeja) [inline public/index.pug]: Let me know when you have implemented the view in detail and I'll continue the review 👍🏼

Comment (jorgeamadosoria) [reply public/index.pug]: will do!

Comment (jorgeamadosoria) [reply public/index.pug]: Icon detail view is now added. 
Set width of 400px; I think this sufficient but there may be other opinions. 
I've restored guidelines, and licenses in this view, so that info that is no longer present in the layouts can still be seen here.
I've also added a color SVG download from this view.

**Comment by mondeja** [comment]
Would you mind to resolve the conflicts @jorgeamadosoria?

**Comment by jorgeamadosoria** [comment]
> Would you mind to resolve the conflicts @jorgeamadosoria?

done

**Comment by jorgeamadosoria** [comment]
All the conflicts are resolved. Can we get this merged or rejected before more conflicts arise, please, @mondeja ?

**Comment by mondeja** [comment]
I'll try to merge this with master ASAP

File: public/index.pug
Comment (mondeja) [inline public/index.pug:389]: Not sure if `boder-` classes are still in the Style?

Comment (mondeja) [reply public/index.pug:389]: No

**Pull request merged**
